scipy/optimize/_linprog.py

```diff
@@ -506,21 +506,18 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
 
     # Keep the original arrays to calculate slack/residuals for original
     # problem.
-    c_o, A_ub_o, b_ub_o, A_eq_o, b_eq_o = c.copy(
-    ), A_ub.copy(), b_ub.copy(), A_eq.copy(), b_eq.copy()
+    lp_o = lp
```

Comment:
Can you explain how you got around making copies here? Were they not necessary?

Reply:
Thanks for the reminder! It took me a while to figure out the best way to do this, but as it turns out `lp_o` is a copy of `lp` because of the immutability property of `namedtuple`.
Instead of having functions modify the value of `c`, `A_ub`, ... etc. in place, I do a `return _LPProblem(c, A_ub, b_ub, A_eq, b_eq, bounds, x0)` so that if debugging is needed in the future, it's easy to compare the before-and-after of elements to spot changes.

I'm suspecting though that I may have missed some of the implicit modifications that could have caused the 6 remaining failed tests.

Reply:
Not exactly. Even if you were to copy the tuple, it doesn't copy the objects the tuple refers to. The tuple only stores references to the objects, but both copies of the tuple would refer to the same objects because you've only copied the references.

Reply:
My apologies for misusing the word "copy"!
What I meant was that `lp_o` is intended to be a preserved reference of `lp` before `lp` gets re-assigned to a new `namedtuple` after `_presolve` (line 515).

```Python
lp_o = lp

# Solve trivial problem, eliminate variables, tighten bounds, etc...
c0 = 0  # we might get a constant term in the objective
if solver_options.pop('presolve', True):
    rr = solver_options.pop('rr', True)
    (lp, c0, x, undo, complete, status, message) = _presolve(lp, rr, tol) # here lp points to a new namedtuple
```